tools/autograd/derivatives.yaml

```diff
@@ -606,13 +606,13 @@
   other: zeros_like(other)
 
 - name: lerp.Scalar(Tensor self, Tensor end, Scalar weight) -> Tensor
-  self: grad * (1 - weight.toDouble())
-  end: grad * weight
+  self: "handle_r_to_c(self.scalar_type(), grad * (1 - (weight.isComplex() ? weight.conj().toComplexDouble() :weight.toDouble())))"
```

Comment:
I don't think we need the `handle_r_to_c` since the inputs to the function require weight to be a float when self and end are floating point tensors.
```
>>> start=torch.randn(4)
>>> end=torch.randn(4)
>>> torch.lerp(start, end, 2j)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: value cannot be converted to type float without overflow: (0,2)
```